src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -78,60 +71,21 @@ const UsersFragment = ({
     Keyboard.dismiss();
   });
 
-  const exhangeFollower = (newUserLists, willFollow, userId) => {
-    const indexUser = newUserLists.findIndex((item) =>
-      item.user ? item.user.user_id === userId : item.user_id === userId
-    );
-    if (newUserLists[indexUser].user) {
-      newUserLists[indexUser].user.following = !!willFollow;
-      // newUserLists[indexUser].user.user_id_follower = myId;
-    } else {
-      newUserLists[indexUser].following = !!willFollow;
-      // newUserLists[indexUser].user_id_follower = myId;
-    }
-    return newUserLists[indexUser];
-  };
-
-  const mapUser = (newUser) => {
-    return discovery.initialUsers.map((user) => {
-      if (user.user) {
-        if (user.user.user_id === newUser.user.user_id) return newUser;
-      } else if (user.user_id === newUser.user_id) return newUser;
-      return user;
-    });
-  };
-
   const handleUser = async (from, willFollow, item) => {
     if (from === FROM_FOLLOWED_USERS_INITIAL || from === FROM_UNFOLLOWED_USERS_INITIAL) {
-      const newFollowedUsers = [...users];
-      const newUser = exhangeFollower(
-        newFollowedUsers,
-        willFollow,
-        item.user ? item.user.user_id : item.user_id
-      );
-
-      DiscoveryAction.setDiscoveryInitialUsers(mapUser(newUser), discoveryDispatch);
+      updateFollowDiscoveryContext(willFollow, item);
     }
 
     if (from === FROM_FOLLOWED_USERS) {
       const newFollowedUsers = [...followedUsers];
       exhangeFollower(newFollowedUsers, willFollow, item.user ? item.user.user_id : item.user_id);
-
       setFollowedUsers(newFollowedUsers);
     }
-
     if (from === FROM_UNFOLLOWED_USERS) {
       const newUnfollowedUsers = [...unfollowedUsers];
       exhangeFollower(newUnfollowedUsers, willFollow, item.user ? item.user.user_id : item.user_id);
       setUnfollowedUsers(newUnfollowedUsers);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactoring done here simplifies the code and makes it more readable. The use of `updateFollowDiscoveryContext` and `exhangeFollower` functions abstracts away some of the complexity. However, make sure these functions are well-tested to avoid potential bugs. Also, note that the handling for `FROM_USERS_INITIAL` has been removed. If this case is no longer relevant, then it's fine. Otherwise, its removal could lead to unexpected behavior.


<!-- This is an auto-generated comment by OSS CodeRabbit -->